packages/playwright/src/reporter.ts

```diff
@@ -129,23 +133,31 @@ class ReplayPlaywrightReporter implements Reporter {
     });
   }
 
-  getFixtureData(test: TestIdData) {
-    const key = this.getTestKey(test);
-    this.fixtureData[key] ??= {
+  getFixtureData(test: TestExecutionIdData) {
+    const id = this._getTestExecutionId(test);
+    this.fixtureData[id] ??= {
       steps: [],
       stacks: {},
       filenames: new Set(),
     };
 
-    return this.fixtureData[key];
-  }
-
-  getTestKey(test: TestIdData) {
-    return [test.id, test.attempt, ...test.source.scope, test.source.title].join("-");
+    return this.fixtureData[id];
   }
 
-  getTestId(test: TestCase) {
-    return test.titlePath().join("-");
+  // Playwright alrady provides a unique test id:
+  // https://github.com/microsoft/playwright/blob/6fb214de2378a9d874b46df6ea99d04da5765cba/packages/playwright/src/common/suiteUtils.ts#L56-L57
+  // this is different because it includes `repeatEachIndex` and `attempt`
+  // TODO: this could be simplified to `${test.testId}-${test.repeatEachIndex}-${test.attempt}`
+  // before doing that all recipients of `TestExecutionIdData` should be rechecked to see if such a change would be safe
```

Comment:
Can you file a Linear follow-up issue for this and update the TODO comment to reference it?

Reply:
done